sonar-db/src/test/java/org/sonar/db/charset/MysqlCharsetHandlerTest.java

```diff
@@ -49,71 +45,54 @@
-    expectedException.expectMessage("UTF8 case-sensitive collation is required for database columns [projects.name]");
-
-    underTest.handle(mock(Connection.class), immutableEnumSet(ENFORCE_UTF8));
-  }
-
-  @Test
-  public void fail_if_not_utf8() throws Exception {
-    answerColumnDef(asList(
-      new ColumnDef(TABLE_ISSUES, COLUMN_KEE, "utf8", "utf8_bin", "varchar", 10, false),
-      new ColumnDef(TABLE_PROJECTS, COLUMN_KEE, "latin1", "latin1_german1_ci", "varchar", 10, false),
-      new ColumnDef(TABLE_PROJECTS, COLUMN_NAME, "latin1", "latin1_swedish_ci", "varchar", 20, false)));
-
-    expectedException.expect(MessageException.class);
-    expectedException.expectMessage("UTF8 case-sensitive collation is required for database columns [projects.kee, projects.name]");
-    underTest.handle(mock(Connection.class), immutableEnumSet(ENFORCE_UTF8));
+  public void fresh_install_does_not_verify_anything() throws Exception {
+    underTest.handle(connection, DatabaseCharsetChecker.State.FRESH_INSTALL);
+    verifyZeroInteractions(sqlExecutor);
   }
 
   @Test
   public void repair_case_insensitive_column() throws Exception {
-    answerColumnDef(asList(
+    answerColumnDef(
       new ColumnDef(TABLE_ISSUES, COLUMN_KEE, "utf8", "utf8_bin", "varchar", 10, false),
-      new ColumnDef(TABLE_PROJECTS, COLUMN_NAME, "latin1", "latin1_swedish_ci", "varchar", 10, false)));
+      new ColumnDef(TABLE_PROJECTS, COLUMN_NAME, "latin1", "latin1_swedish_ci", "varchar", 10, false));
 
-    Connection connection = mock(Connection.class);
-    underTest.handle(connection, immutableEnumSet(AUTO_REPAIR_COLLATION));
+    underTest.handle(connection, DatabaseCharsetChecker.State.UPGRADE);
 
-    verify(selectExecutor).executeUpdate(connection, "ALTER TABLE projects MODIFY name varchar(10) CHARACTER SET 'latin1' COLLATE 'latin1_bin' NOT NULL");
+    verify(sqlExecutor).executeUpdate(connection, "ALTER TABLE projects MODIFY name varchar(10) CHARACTER SET 'latin1' COLLATE 'latin1_bin' NOT NULL");
   }
 
   @Test
   public void size_should_be_ignored_on_longtext_column() throws Exception {
-    answerColumnDef(asList(new ColumnDef(TABLE_ISSUES, COLUMN_KEE, "latin1", "latin1_german1_ci", "longtext", 4_294_967_295L, false)));
+    answerColumnDef(
+      new ColumnDef(TABLE_ISSUES, COLUMN_KEE, "latin1", "latin1_german1_ci", "longtext", 4_294_967_295L, false));
 
-    Connection connection = mock(Connection.class);
-    underTest.handle(connection, immutableEnumSet(AUTO_REPAIR_COLLATION));
+    underTest.handle(connection, DatabaseCharsetChecker.State.UPGRADE);
 
-    verify(selectExecutor).executeUpdate(connection, "ALTER TABLE " + TABLE_ISSUES + " MODIFY " + COLUMN_KEE + " longtext CHARACTER SET 'latin1' COLLATE 'latin1_bin' NOT NULL");
+    verify(sqlExecutor).executeUpdate(connection, "ALTER TABLE " + TABLE_ISSUES + " MODIFY " + COLUMN_KEE + " longtext CHARACTER SET 'latin1' COLLATE 'latin1_bin' NOT NULL");
   }
 
   @Test
   public void tests_toCaseSensitive() {
```

Comment:
it this test, which required to change visibility of method `toCaseSensitive` really required?
it is tested in `repair_case_insensitive_column` (with `latin1`, agreed, but the test is effectively the same)